src/Common/Core/Impl/OS/Win32Process.cs

```diff
@@ -117,43 +129,48 @@ public class Win32Process {
             SECURITY_ATTRIBUTES processAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
             SECURITY_ATTRIBUTES threadAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
 
-            PROCESS_INFORMATION pi;
-            try {
-                if(winIdentity == null) {
-                    if(!CreateProcess(applicationName, commandLine, ref processAttr, ref threadAttr, true,
-                        (uint)(CREATE_PROCESS_FLAGS.CREATE_UNICODE_ENVIRONMENT | CREATE_PROCESS_FLAGS.CREATE_NO_WINDOW),
-                        environment.NativeEnvironmentBlock,
-                        workingDirectory,
-                        ref si,
-                        out pi)) {
-                        throw new Win32Exception(Marshal.GetLastWin32Error());
+            lock (_createProcessLock) {
```

Comment:
SetErrorMode is needed to prevent system from showing a message box if a critical error occurs.